package gtkclient;

message po8eCard {
	required uint64 id = 1; // zero-indexed
	repeated po8eChannel channel = 2;
}

message po8eChannel {
	required uint64 id = 1; // 1-indexed
	optional string name = 2;
	optional uint64 scale_factor = 3 [default = 1];
	enum DataTypes {
	  NEURAL = 0; // uses scale_factor
	  EVENT = 1;
	  ANALOG = 2; // uses scale_factor
	  STIM_PULSES = 3;
	  BLANK_CLOCK = 4;
	}
	optional DataTypes data_type = 4 [default = NEURAL];
}
